improver_tests/calibration/reliability_calibration/test_ConstructRealizationCalibrationTables.py

```diff
@@ -0,0 +1,366 @@
+                                         [[0., 0., 0.],
+                                          [0., 0., 0.],
+                                          [0.75, 0.875, 0.]],
+                                         [[0., 0., 0.],
+                                          [0., 0., 0.],
+                                          [0., 0., 1.]]],
+                                        [[[1., 0., 0.],
+                                          [0., 0., 0.],
+                                          [0., 0., 0.]],
+                                         [[0., 1., 1.],
+                                          [0., 0., 0.],
+                                          [0., 0., 0.]],
+                                         [[0., 0., 0.],
+                                          [1., 1., 1.],
+                                          [0., 0., 0.]],
+                                         [[0., 0., 0.],
+                                          [0., 0., 0.],
+                                          [1., 1., 0.]],
+                                         [[0., 0., 0.],
+                                          [0., 0., 0.],
+                                          [0., 0., 1.]]]], dtype=np.float32)
+
+
+class Test__init__(unittest.TestCase):
+
+    """Test the __init__ method."""
+
+    def test_using_defaults(self):
+        """Test without providing any arguments."""
+        plugin = Plugin()
+        self.assertEqual(len(plugin.probability_bins), 5)
+        self.assertEqual(plugin.expected_table_shape, (3, 5))
+
+    def test_with_arguments(self):
+        """Test with specified arguments."""
+        plugin = Plugin(n_probability_bins=4, single_value_limits=False)
+        self.assertEqual(len(plugin.probability_bins), 4)
+        self.assertEqual(plugin.expected_table_shape, (3, 4))
+
+
+class Test__repr__(unittest.TestCase):
+
+    """Test the __repr__ method."""
+
+    def test_basic(self):
+        """Test repr is as expected."""
+        plugin = Plugin(n_probability_bins=2, single_value_limits=False)
+        self.assertEqual(
+            str(plugin),
+            '<ConstructRealizationCalibrationTables: probability_bins: '
+            '[0.00 --> 0.50], [0.50 --> 1.00]>')
+
+
+class Test__define_probability_bins(unittest.TestCase):
+
+    """Test the _define_probability_bins method."""
+
+    def test_without_single_value_limits(self):
+        """Test the generation of probability bins without single value end
+        bins. The range 0 to 1 will be divided into 4 equally sized bins."""
+        expected = np.array(
+            [[0., 0.24999999],
+             [0.25, 0.49999997],
+             [0.5, 0.74999994],
+             [0.75, 1.]])
+        result = Plugin._define_probability_bins(n_probability_bins=4,
+                                                 single_value_limits=False)
+        assert_almost_equal(result, expected)
+
+    def test_with_single_value_limits(self):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=4869046)